Server/Services/ILocalTempFileLocks.cs

```diff
@@ -14,15 +16,44 @@ public interface ILocalTempFileLocks
     ///   to avoid multiple places using the same temporary folder at once
     /// </summary>
     /// <param name="suffix">Suffix to add to the temporary folder</param>
-    /// <param name="path">The final result path</param>
-    /// <returns>A semaphore that needs to be locked while using the path</returns>
-    public SemaphoreSlim GetTempFilePath(string suffix, out string path);
+    /// <returns>The final result path</returns>
+    public string GetTempFilePath(string suffix);
+
+    /// <summary>
+    /// Asynchronously lock based on a key.
```

Comment:
XML content should be intended by extra 2 spaces for each nesting level (due to a bug in the checking tools this is not currently automatically enforced).

```suggestion
    ///   Asynchronously lock based on a key.
```

Reply:
Done, not yet committed.